File: 1752. check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        count, n = 0, len(nums)
        for i in range(n):
            if nums[i] > nums[(i+1) % n]:
                count += 1
                if count > 1: return False
        return True

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1
    nums1 = [3,4,5,1,2]
    output1 = sol.check(nums1)
    print(f"Example 1: nums={nums1}")
    print(f"Output: {output1}")
    print(f"Expected: True")
    print()
    
    # Example 2
    nums2 = [2,1,3,4]
    output2 = sol.check(nums2)
    print(f"Example 2: nums={nums2}")
    print(f"Output: {output2}")
    print(f"Expected: False")
    print()
    
    # Example 3
    nums3 = [1,2,3]
    output3 = sol.check(nums3)
    print(f"Example 3: nums={nums3}")
    print(f"Output: {output3}")
    print(f"Expected: True")

File: 1752. check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int breaks = 0;
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            if (nums[i] > nums[(i + 1) % n]) {
                breaks++;
            }
        }

        return breaks <= 1;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1
        int[] nums1 = {3,4,5,1,2};
        boolean output1 = sol.check(nums1);
        System.out.println("Example 1: nums=[3,4,5,1,2]");
        System.out.println("Output: " + output1);
        System.out.println("Expected: true");
        System.out.println();
        
        // Example 2
        int[] nums2 = {2,1,3,4};
        boolean output2 = sol.check(nums2);
        System.out.println("Example 2: nums=[2,1,3,4]");
        System.out.println("Output: " + output2);
        System.out.println("Expected: false");
        System.out.println();
        
        // Example 3
        int[] nums3 = {1,2,3};
        boolean output3 = sol.check(nums3);
        System.out.println("Example 3: nums=[1,2,3]");
        System.out.println("Output: " + output3);
        System.out.println("Expected: true");
    }
}